galsim/catalog.py

```diff
@@ -312,3 +317,334 @@ def itervalues(self):
+def _makeReal(cat, index, pad_size):
+    noise_pad_size = int(np.ceil(pad_size * np.sqrt(2.)))
+    gal = galsim.RealGalaxy(cat, index=index, noise_pad_size=noise_pad_size)
+
+    # Rescale the galaxy size.
+    if hasattr(cat, 'size_factor'):
+        gal.applyDilation(cat.size_factor)
+    # Rescale the galaxy flux.
+    if hasattr(cat, 'flux_factor'):
+        gal *= cat.flux_factor
+    return gal
+
+# This will act essentially as a static variable for the _makeParam function
+_COSMOS_bandpass = None
+
+def _makeParam(cat, index, chromatic=False):
+    if _COSMOS_bandpass is None:
+        # Defer making the Bandpass until we actually use it.
+        # It's not a huge calculation, but the thin() call especially isn't trivial.
+        _COSMOS_bandpass = galsim.Bandpass(
+                os.path.join(galsim.meta_data.share_dir, 'wfc_F814W.dat.gz'),
+                wave_type='ang').thin().withZeropoint(25.94)
+
+    record = cat[index]
+
+    if chromatic:
+        # Read in some SEDs.
+        sed_bulge = galsim.SED(os.path.join(galsim.meta_data.share_dir,'CWW_E_ext.sed'))
+        sed_disk = galsim.SED(os.path.join(galsim.meta_data.share_dir,'CWW_Scd_ext.sed'))
+        sed_intermed = galsim.SED(os.path.join(galsim.meta_data.share_dir,'CWW_Sbc_ext.sed'))
+
+    # Get fit parameters.
+    params = record.field('bulgefit')
+    sparams = record.field('sersicfit')
+    # Get the status flag for the fits.
+    bstat = record.field('fit_status')[0]
+    sstat = record.field('fit_status')[4]
+    # Get the precomputed bulge-to-total ratio for the 2-component fits.
+    dvc_btt = record.field('fit_dvc_btt')
+    # Get the precomputed median absolute deviation for the 1- and 2-component fits.
+    bmad = record.field('fit_mad_b')
+    smad = record.field('fit_mad_s')
+
+    # First decide if we can / should use bulgefit, otherwise sersicfit.  This decision process
+    # depends on: the status flags for the fits, the bulge-to-total ratios (if near 0 or 1, just use
+    # single component fits), the sizes for the bulge and disk (if <=0 then use single component
+    # fits), the axis ratios for the bulge and disk (if <0.051 then use single component fits), and
+    # a comparison of the median absolute deviations to see which is better.
+    use_bulgefit = 1
+    if bstat<1 or bstat>4 or dvc_btt<0.1 or dvc_btt>0.9 or np.isnan(dvc_btt) or params[9]<=0 or \
+            params[1]<=0 or params[11]<0.051 or params[3]<0.051 or smad<bmad:
+        use_bulgefit = 0
+        # Then check if sersicfit is viable; if not, this object is a total failure:
+        if sstat<1 or sstat>4 or sparams[1]<=0 or sparams[0]<=0:
+            raise RuntimeError("Cannot make parametric model for this object")
+
+    # Now, if we're supposed to use the 2-component fits, get all the parameters.
+    if use_bulgefit:
+        # Bulge parameters.
+        bulge_q = params[11]
+        bulge_beta = params[15]*galsim.radians
+        bulge_hlr = 0.03*np.sqrt(bulge_q)*params[9]
```

Comment:
Try to avoid magic numbers as much as possible.  Should name this one `cosmos_pixel_scale = 0.03`.